### PR TITLE
Fix: #67 - Supporting custom probes

### PR DESCRIPTION
**Pull request by ricardozanini**
See #67

### CODE REVIEW

File: hack/ci/operator-olm-test.sh [hunk 1]
```diff
@@ -29,6 +30,12 @@ if [ -z ${KUBECONFIG} ]; then
     ls -la ${KUBECONFIG}
 fi
 
+csv_file=${OUTPUT}/nexus-operator-m88i/${OP_VERSION}/nexus-operator.v${OP_VERSION}.clusterserviceversion.yaml
+echo "---> Updating CSV file '${csv_file}' to imagePullPolicy: Never"
+sed -i 's/imagePullPolicy: Always/imagePullPolicy: Never/g' ${csv_file}
+echo "---> Resulting imagePullPolicy"
+cat ${csv_file} | grep -irn imagePullPolicy
```
Comment (ricardozanini) [inline hack/ci/operator-olm-test.sh:37]:
```suggestion
echo "---> Resulting imagePullPolicy on manifes files"
grep -rn imagePullPolicy ${OUTPUT}/nexus-operator-m88i
```

File: pkg/controller/nexus/resource/deployment.go [hunk 2]
```diff
@@ -117,12 +120,38 @@ func addProbes(deployment *appsv1.Deployment) {
...
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.FailureThreshold, 1)
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.InitialDelaySeconds =
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.InitialDelaySeconds, 0)
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.PeriodSeconds =
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.PeriodSeconds, 1)
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.SuccessThreshold =
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.SuccessThreshold, 1)
+		deployment.Spec.Template.Spec.Containers[0].ReadinessProbe.TimeoutSeconds =
+			util.EnsureMinimum(nexus.Spec.ReadinessProbe.TimeoutSeconds, 1)
+	}
```
[the remark this answers is not in the document]
Comment (ricardozanini) [reply pkg/controller/nexus/resource/deployment.go:154]: We don't need to validate in the Reconcile Loop since we're enforcing this check in the OpenAPI:  https://github.com/m88i/nexus-operator/pull/68/files#diff-115e08ad98fca2d2d354114263813118R139

(this is where we will have calls in the loop)

By adding this check on deployment creation, I can enforce the check at the API level in case we have an external call from other functions/internal packages. This is not an interface validation, but a sanity check for the k8s API integration.

Having a TLS check in the reconcile loop makes sense because we can't enforce that in the OpenAPI (external interface) level. But I think we could add this check in the resource creation as well (or better, move it to there).

To sum up, interface checks we do in the reconcile, internal calls validations we do in the package/internal level.

[the remark this answers is not in the document]
Comment (LCaparelli) [reply pkg/controller/nexus/resource/deployment.go:154]: I see, that makes sense! I hadn't taken into account API usage, I was only considering usage from the reconcile loop point of view.

File: hack/ci/operator-olm-test.sh [hunk 3]
```diff
@@ -33,8 +33,8 @@ fi
 csv_file=${OUTPUT}/nexus-operator-m88i/${OP_VERSION}/nexus-operator.v${OP_VERSION}.clusterserviceversion.yaml
 echo "---> Updating CSV file '${csv_file}' to imagePullPolicy: Never"
 sed -i 's/imagePullPolicy: Always/imagePullPolicy: Never/g' ${csv_file}
-echo "---> Resulting imagePullPolicy"
-cat ${csv_file} | grep -irn imagePullPolicy
+echo "---> Resulting imagePullPolicy on manifest files"
+grep -rn imagePullPolicy ${OUTPUT}/nexus-operator-m88i
```
Comment (LCaparelli) [inline hack/ci/operator-olm-test.sh:37]: I forgot to comment this one on my previous review, it makes sense to me (reducing.the.amount.of.processes++)
